polaris/dataset/_base.py

```diff
@@ -123,14 +122,6 @@ def _validate_base_dataset_model(cls, m: "BaseDataset"):
                 m.annotations[c] = ColumnAnnotation()
             m.annotations[c].dtype = m.dtypes[c]
 
-        #
-        # Set the default cache dir if none and make sure it exists
-        if m.cache_dir is None:
```

Comment:
Why'd we move this back out of`BaseDataset` into the separate dataset classes again?

Reply:
This is related to your other comment: Since the `BaseDataset` no longer inherits from the `ChecksumMixin`, the `_md5sum` property may not exist for subclasses.